xcp_d/utils/dcan2fmriprep.py

```diff
@@ -11,315 +12,411 @@
+            anat_dir_fmriprep, f"{sub_id}_{ses_id}_desc-preproc_T1w.nii.gz"
+        )
+        copy_dictionary[t1w_orig] = [t1w_fmriprep]
 
-        t1wim = anatdir + sub_id + "_" + sess + "_desc-preproc_T1w.nii.gz"
-        t1seg = anatdir + sub_id + "_" + sess + "_dseg.nii.gz"
-        t1ribbon = anatdir + sub_id + "_" + sess + "_desc-ribbon.nii.gz"
-        t1brainm = anatdir + sub_id + "_" + sess + "_desc-brain_mask.nii.gz"
-        regfile1 = (
-            anatdir + sub_id + "_" + sess + "_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5"
+        # NOTE: We're using the T1w image as a transform. This doesn't make sense.
+        t1w_to_template_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5",
         )
-        regfile2 = (
-            anatdir + sub_id + "_" + sess + "_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5"
+        copy_dictionary[t1w_orig].append(t1w_to_template_fmriprep)
+
+        # NOTE: We're using the T1w image as a transform. This doesn't make sense.
+        template_to_t1w_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5",
         )
+        copy_dictionary[t1w_orig].append(template_to_t1w_fmriprep)
 
-        lMid = anatdir + sub_id + "_" + sess + "_hemi-L_midthickness.surf.gii"
-        rMid = anatdir + sub_id + "_" + sess + "_hemi-R_midthickness.surf.gii"
-
-        lpial = anatdir + sub_id + "_" + sess + "_hemi-L_pial.surf.gii"
-        rpial = anatdir + sub_id + "_" + sess + "_hemi-R_pial.surf.gii"
-
-        lwhite = anatdir + sub_id + "_" + sess + "_hemi-L_smoothwm.surf.gii"
-        rwhite = anatdir + sub_id + "_" + sess + "_hemi-R_smoothwm.surf.gii"
-
-        linf = anatdir + sub_id + "_" + sess + "_hemi-L_inflated.surf.gii"
-        rinf = anatdir + sub_id + "_" + sess + "_hemi-R_inflated.surf.gii"
-
-        newanatfiles = [
-            t1wim,
-            t1seg,
-            t1ribbon,
-            t1brainm,
-            regfile1,
-            regfile2,
-            lMid,
-            rMid,
-            lpial,
-            rpial,
-            lwhite,
-            rwhite,
-            linf,
-            rinf,
-        ]
+        brainmask_orig = os.path.join(anat_dir_orig, "brainmask_fs.nii.gz")
+        brainmask_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_desc-brain_mask.nii.gz",
+        )
+        copy_dictionary[brainmask_orig] = [brainmask_fmriprep]
 
-        for i, j in zip(dcanimages, newanatfiles):
-            copyfileobj_example(i, j)
+        # NOTE: What is this file for?
+        ribbon_orig = os.path.join(anat_dir_orig, "ribbon_orig.nii.gz")
+        ribbon_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_desc-ribbon_T1w.nii.gz",
+        )
+        copy_dictionary[ribbon_orig] = [ribbon_fmriprep]
```

Comment:
Needs the space entity, but Anders will look into whether we need this file.

Reply:
This ribbon file is a binary volume mask of the cortical ribbon where each voxel is either a 1 or 0 depending upon whether it falls in the ribbon or not. It's used to construct the volume to surface mapping.

Reply:
Since we're not going to generate a brainsprite when surfaces aren't already available, I don't think we actually need to map volumes to surfaces anymore. @ericfeczko @madisoth does that sound right? Can we drop the ribbon file?